pkg/controller/remotemachineset/gcpactuator.go

```diff
@@ -244,6 +241,42 @@ func (a *GCPActuator) obtainLease(pool *hivev1.MachinePool, cd *hivev1.ClusterDe
 		if err != nil {
 			return "", false, err
 		}
+		if len(availLeaseChars) == 0 {
+			logger.Warn("no GCP MachinePoolNameLease characters available, setting condition")
+			conds, changed := controllerutils.SetMachinePoolConditionWithChangeCheck(
+				pool.Status.Conditions,
+				hivev1.NoMachinePoolNameLeasesAvailable,
+				corev1.ConditionTrue,
+				"OutOfMachinePoolNames",
+				"All machine pool names are in use",
+				controllerutils.UpdateConditionIfReasonOrMessageChange,
+			)
+			if changed {
+				pool.Status.Conditions = conds
+				if err := a.client.Status().Update(context.Background(), pool); err != nil {
+					logger.WithError(err).Log(controllerutils.LogLevel(err), "failed to update MachinePool conditions")
```

Comment:
An error should be returned here. Also, logging here is redundant with the logging that will be done in the caller.